apps/fft/fft-serial.cpp

```diff
@@ -30,6 +30,67 @@
 
 #include "fft.hpp"
 
+//
+// serial fft function
+//
+std::vector<data_t> fft_serial(data_t *x, const int N, bool debug = false)
```

Comment:
suggest to add `[[nodiscard]]` to enforce the fact that we cannot ignore return value from this function

Reply:
if we are not updating x, apply `const` for protection.